### PR TITLE
EG-1227, restructure code and tests into app and app/core

### PR DESCRIPTION
**Pull request by cmrhmrc**

### CODE REVIEW

File: app/core/models/ocelot/errors/GuidanceErrors.scala [hunk 1]
```diff
@@ -14,7 +14,7 @@
  * limitations under the License.
  */
 
-package models.ocelot.errors
+package core.models.ocelot.errors
```
Comment (woollett-r) [inline app/core/models/ocelot/errors/GuidanceErrors.scala:17]: IntelliJ is reporting a typo at line 39: InconsistenQuestionError

Comment (cmrhmrc) [reply app/core/models/ocelot/errors/GuidanceErrors.scala:17]: fixed

File: app/services/ScratchService.scala [hunk 2]
```diff
@@ -36,7 +36,7 @@ class ScratchService @Inject() (repository: ScratchRepository,
   val logger = Logger(getClass)
 
   def save(json: JsObject): Future[RequestOutcome[UUID]] =
-    guidancePages(pageBuilder, securedProcessBuilder, json).fold(
+    guidancePages(pageBuilder, json)(c, securedProcessBuilder).fold(
```
Comment (woollett-r) [inline app/services/ScratchService.scala:39]: I think "appConfig" is better than "c".

File: app/services/package.scala [hunk 3]
```diff
@@ -14,31 +14,38 @@
...
+          pages => Right((p, pages, js))
         )
       }
     )
 
-  def fakeWelshText(process: Process)(implicit c: AppConfig): Process =
-    if (!process.passPhrase.isEmpty || c.fakeWelshInUnauthenticatedGuidance)
-      process.copy(phrases = process.phrases.map(p => if (p.welsh.trim.isEmpty) Phrase(p.english, s"Welsh, ${p.english}") else p))
-    else process
+  private [services] def fakeWelshTextIfRequired(process: Process, jsObject: JsObject)(implicit c: AppConfig): (Process, JsObject) =
```
Comment (woollett-r) [inline app/services/package.scala:40]: For some reason IntelliJ whinges about the space between private and [service]

File: app/services/package.scala [hunk 4]
```diff
@@ -14,31 +14,38 @@
...
+    if (!process.passPhrase.isEmpty || c.fakeWelshInUnauthenticatedGuidance) {
```
Comment (woollett-r) [inline app/services/package.scala:41]: According to IntelliJ you can use process.passPhrase.isDefined here.

File: app/services/package.scala (hunk 3 above)
Comment (cmrhmrc) [reply app/services/package.scala:40]: its doesnt like the space between privat and [services]

File: app/services/package.scala (hunk 4 above)
Comment (cmrhmrc) [reply app/services/package.scala:41]: yes

File: app/services/ScratchService.scala (hunk 2 above)
Comment (cmrhmrc) [reply app/services/ScratchService.scala]: removed unecessary implicit arg lists

**Pull request merged**